pkg/webhook/ippool/validator.go

```diff
@@ -0,0 +1,237 @@
+package ippool
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/containernetworking/plugins/plugins/ipam/host-local/backend/allocator"
+	"github.com/harvester/webhook/pkg/server/admission"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	lbv1 "github.com/harvester/harvester-load-balancer/pkg/apis/loadbalancer.harvesterhci.io/v1beta1"
+	ctllbv1 "github.com/harvester/harvester-load-balancer/pkg/generated/controllers/loadbalancer.harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester-load-balancer/pkg/ipam"
+	"github.com/harvester/harvester-load-balancer/pkg/utils"
+)
+
+type ipPoolValidator struct {
+	admission.DefaultValidator
+	ipPoolCache ctllbv1.IPPoolCache
+}
+
+var _ admission.Validator = &ipPoolValidator{}
+
+func NewIPPoolValidator(ipPoolCache ctllbv1.IPPoolCache) admission.Validator {
+	return &ipPoolValidator{
+		ipPoolCache: ipPoolCache,
+	}
+}
+
+func (i *ipPoolValidator) Create(_ *admission.Request, newObj runtime.Object) error {
+	pool := newObj.(*lbv1.IPPool)
+	if len(pool.Spec.Ranges) == 0 {
+		return fmt.Errorf(createErr, pool.Name, fmt.Errorf("range could not be empty"))
```

Comment:
error type returned here should be AdmitError

Reply:
https://github.com/rancher/wrangler/blob/39a4707f0689cc97ca955f5eea5d0b37e1664cae/pkg/webhook/router.go#L39

Reply:
```
func (r *Router) sendError(rw http.ResponseWriter, review *v1.AdmissionReview, err error) {
	logrus.Error(err)
	if review == nil || review.Request == nil {
		http.Error(rw, err.Error(), http.StatusInternalServerError)
		return
	}
	review.Response.Result = &errors.NewInternalError(err).ErrStatus
	writeResponse(rw, review)
}
```
It still returns internal error from wrangler even if we return AdmitError.

Reply:
we can ignore this issue in this PR, consider add an enhancement in next release